include/counter.h

```diff
@@ -100,26 +172,76 @@ static inline u32_t _impl_counter_read(struct device *dev)
 
 /**
  * @brief Set an alarm.
- * @param dev Pointer to the device structure for the driver instance.
- * @param callback Pointer to the callback function. If this is NULL,
- *                 this function unsets the alarm.
- * @param count Number of counter ticks. This is relative value, meaning
- *              an alarm will be triggered count ticks in the future.
- * @param user_data Pointer to user data.
+ *
+ * In case of absolute request, maximal value that can be set is equal to
+ * wrap value set by counter_set_wrap call or default, maximal one. in case of
+ * relative request, Maximal value can be retrieved using
+ * counter_get_max_relative_alarm.
+ *
+ * @param dev	Pointer to the device structure for the driver instance.
+ * @param cb	Alarm callback.
  *
  * @retval 0 If successful.
  * @retval -ENOTSUP if the counter was not started yet.
  * @retval -ENODEV if the device doesn't support interrupt (e.g. free
  *                        running counters).
+ * @retval -ENOTSUP if request is not supported.
+ * @retval -EINVAL if alarm settings are invalid.
  * @retval Negative errno code if failure.
  */
 static inline int counter_set_alarm(struct device *dev,
-				    counter_callback_t callback,
-				    u32_t count, void *user_data)
+				    const struct counter_alarm_cfg *alarm_cfg)
+{
+	const struct counter_driver_api *api = dev->driver_api;
+
+	return api->set_alarm(dev, alarm_cfg);
+}
+
+/**
+ * @brief Disable an alarm.
+ *
+ * @param dev	Pointer to the device structure for the driver instance.
+ * @param cb	Alarm callback.
+ *
+ * @retval 0 If successful.
+ * @retval -ENOTSUP if the counter was not started yet.
+ * @retval -ENOTSUP if request is not supported.
+ * @retval Negative errno code if failure.
```

Comment:
Is it necessary to add it after all possible error codes were listed? Same story with `set_alarm` and `set_wrap_alarm`.

Reply:
removed